From b38e96adb3b40c7b3e0da011cf065a4681dae494 Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Tue, 16 Sep 2014 12:07:31 +0300
Subject: [PATCH 069/399] dpa_offload: Avoid leaking memory for replicator
 params

In function "dpa_classif_mcast_create_group_error" in the dpa_classifier
driver, the allocated memory for "replic_grp_params" is leaking in case
of errors. This change adds a release of this memory in case it is
allocated when the function returns an error.

Change-Id: Ic0f537f3e980d498f11ecf6c2874674b7003b2a9
Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
Reviewed-on: http://git.am.freescale.net:8181/18809
Reviewed-by: Radu-Andrei Bulie <Radu.Bulie@freescale.com>
[Lu:Original patch taken from
QorIQ-SDK-V1.7-SOURCE-20141218-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_classifier.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_classifier.c b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
index f4760fa..fdd0c45 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_classifier.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
@@ -7968,6 +7968,7 @@ dpa_classif_mcast_create_group_error:
 		}
 		kfree(pgroup);
 	}
+	kfree(replic_grp_params);
 
 	*grpd = DPA_OFFLD_DESC_NONE;
 
-- 
1.7.5.4

